product_extended_segmentation/model/template.py

```diff
@@ -90,7 +90,10 @@ def _calc_price(
         wizard_obj = self.pool.get("stock.change.standard.price")
         bom_obj = self.pool.get('mrp.bom')
         prod_obj = self.pool.get('product.product')
-
+        user = self.pool.get('res.users').browse(cr, uid, uid, context)
+        bottom_price_threshold = user.company_id.std_price_neg_threshold
```

Comment:
company of product or company of user.

First priority the company_id on product itself to take the value of thershold